scripts/sct_extract_metric.py

```diff
@@ -55,169 +57,213 @@ def get_parser():
-                      description="1: display on, 0: display off (default)",
-                      mandatory=False,
-                      example=["0", "1"],
-                      default_value="1")
-    parser.usage.addSection("\nFOR ADVANCED USERS")
-    parser.add_option(name='-param',
-                      type_value='str',
-                      description="""Advanced parameters for the 'map' method. Separate with comma. All items must be listed (separated with comma).
-    #1: standard deviation of metrics across labels
-    #2: standard deviation of the noise (assumed Gaussian)""",
-                      mandatory=False)
-    parser.add_option(name='-p',
-                      type_value=None,
-                      description="""Advanced parameters for the 'map' method. Separate with comma. All items must be listed (separated with comma).
-    #1: standard deviation of metrics across labels
-    #2: standard deviation of the noise (assumed Gaussian)""",
-                      mandatory=False,
-                      deprecated_by='-param')
-    parser.add_option(name='-fix-label',
-                      type_value=[[','], 'str'],
-                      description='When using ML or MAP estimations, if you do not want to estimate the metric in one label and fix its value to avoid effects on other labels, specify <label_ID>,<metric_value. Example to fix the CSF value to 0: -fix-label 36,0.',
-                      mandatory=False,
-                      default_value='')
-    parser.add_option(name='-norm-file',
-                      type_value='image_nifti',
-                      description='Filename of the label by which the user wants to normalize',
-                      mandatory=False)
-    parser.add_option(name='-n',
-                      type_value='image_nifti',
-                      description='Filename of the label by which the user wants to normalize',
-                      mandatory=False,
-                      deprecated_by='-norm-file')
-    parser.add_option(name='-norm-method',
-                      type_value='multiple_choice',
-                      description='Method to use for normalization:\n- sbs: normalization slice-by-slice\n- whole: normalization by the metric value in the whole label for all slices.',
-                      example=['sbs', 'whole'],
-                      mandatory=False)
-    parser.add_option(name='-mask-weighted',
-                      type_value='image_nifti',
-                      description='Nifti mask to weight each voxel during ML or MAP estimation.',
-                      example='PAM50_wm.nii.gz',
-                      mandatory=False)
-    parser.add_option(name='-discard-neg-val',
-                      type_value='multiple_choice',
-                      mandatory=False,
-                      description='Discard voxels with negative value when computing metrics statistics.',
-                      example=["0", "1"],
-                      default_value="0")
```

Comment:
These old argument names have hyphens (`-`) which isn't allowed in argparse. [A possible workaround](https://stackoverflow.com/a/12834986) is possible, but it might be better to just switch to underscores.

The trouble comes with any existing users who are using the hyphenated names. It's similar to the deprecated names, in that this could be a breaking change.

Reply:
i'm not sure i understand the problem. For example, `-discard-neg-val` would be interpreted in the code as `arguments.discard_neg_val`, wouldn't it?

Reply:
Aha, you're right. My apologies.